ansible_base/rbac/service_api/views.py

```diff
@@ -100,6 +100,52 @@ def _unassign(self, request):
         self.remote_secondary_sync_unassignment(role_definition, actor, content_object, from_service=serializer.validated_data.get('from_service'))
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    def _object_delete(self, request):
+        """Delete ALL role assignments for a specific object"""
+        from ..models import DABContentType, ObjectRole
+        
+        # Validate required fields
+        data = request.data
+        if 'resource_type' not in data or 'resource_pk' not in data:
+            return Response(
+                {'error': 'resource_type and resource_pk are required'}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        
+        try:
+            # Get content type for the resource
+            app_label, model = data['resource_type'].split('.')
+            content_type = DABContentType.objects.get(app_label=app_label, model=model)
+            
+            # Delete ALL assignments for this object (bulk operation)
+            with transaction.atomic():
+                deleted_count = self.get_queryset().filter(
+                    content_type=content_type,
+                    object_id=data['resource_pk']
+                ).delete()[0]
```

Comment:
I think the other signals didn't do this, taking faith in the cascade deletion working

https://github.com/ansible/django-ansible-base/blob/devel/ansible_base/rbac/triggers.py#L260

But whether you do or don't, it may be equally efficient.